src/lib/crypto/DY.Lib.Crypto.AEAD.Split.fst

```diff
@@ -0,0 +1,79 @@
+module DY.Lib.Crypto.AEAD.Split
+
+open Comparse // for_allP, for_allP_eq
+open DY.Core
+open DY.Lib.SplitFunction
+
+let split_aead_predicate_params (cusages:crypto_usages) = {
```

Comment:
I think, we decided to give explicit types for this kind of "instance" definitions, e.g., `: split_function_parameters` should be added here (and in all other cases).

Reply:
Ah indeed!

I am not using a `val` & `let` definition here, because if I define it like this:
```fstar
val split_aead_predicate_params: crypto_usages -> split_function_parameters
let split_aead_predicate_params cusages = {
```
then we get the bogus error
```
- Expected type split_function_parameters but ... has type split_function_parameters
- Computed type split_function_parameters and effect Tot is not compatible with the annotated type split_function_parameters and effect Tot
```
Looks like an F* bug, that I might report if I am not too lazy to minimize.

Reply:
Done